src/parser.ml

```diff
@@ -321,14 +321,14 @@ let parser path = m:path_elem ms:{"." path_elem}* $ -> m::ms
 let parser qident = mp:{path_elem "."}* id:any_ident -> in_pos _loc (mp,id)
 
 (** [symtag] parses a single symbol tag. *)
-let parser symtag =
-  | _const_ -> Sym_const
-  | _inj_   -> Sym_inj
+let parser property =
+  | _constant_  -> Terms.Constant
+  | _injective_ -> Terms.Injective
 
 (** [exposition] parses the exposition tag of a symbol.*)
 let parser exposition =
-  | _protected_ -> Symex_protected
-  | _private_   -> Symex_private
```

Comment:
I think we should only have one syntactic entity for "modifiers", including all of these properties. I originally intended for that kind of extension (hence the star that was on the `symtag` grammar).  We should have a rule of the form:
```ocaml
ms:modifier* _symbol_ s:ident al:arg* ":" a:term
```
And a corresponding type of modifiers in `src/syntax.ml` that contains all the constructors.

Reply:
I see no good reason at the moment to mix visibility markers that could apply to objects that are not symbols, and property markers that apply on function symbols only. And currently there is no need for other visibility/property markers.

Reply:
I was thinking of always allowing them at parse time, but reporting an error when they are used in a way that is not allowed when handling the command. That way an error message could explain why some specific modifier is not allowed in certain places.

Reply:
I understand but I see no necessity at the moment.